hybrasyl/Objects/User.cs

```diff
@@ -1247,12 +1247,48 @@ public void SendVisibleGold(Gold gold)
 
         internal void UseSkill(byte slot)
         {
+            if(!Map.AllowCasting)
+            {
+                if (!IsPrivileged)
+                {
+                    SendSystemMessage("You can't use that here.");
+                    return;
+                }
+            }
             var bookSlot = SkillBook[slot];
             if (bookSlot.OnCooldown)
             {
                 SendSystemMessage("You must wait longer to use that.");
                 return;
             }
+
+            foreach (var cost in bookSlot.Castable.CastCosts)

```

Comment:
this is a duplicate of the check done in `ProcessCastingCost`, which is used by `UseCastable`